### PR TITLE
Add a separate module for spring-cloud-azure-native-reachability

### PR DESCRIPTION
**Pull request by saragluna**
# Description

The Spring AOT support was removed in https://github.com/Azure/azure-sdk-for-java/pull/32742 because we have yet to be ready to GA the AOT support. This PR will introduce another module, the `spring-cloud-azure-native-reachabitity` to host the support for Spring AOT. This module won't be GA in the feature because once we are ready for the Spring AOT support, these codes will be incorporated into each module. 

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by saragluna — COMMENTED** [review]
Two improvements need to address:
- There should only be an entry for Spring readme files in the cspell.json
- To improve the `custom.security` configuration for Maven and Gradle, for right now the two are not in the same folder.

File: sdk/spring/ci.yml [hunk 1]
```diff
@@ -83,6 +83,10 @@ parameters:
   displayName: 'spring-cloud-azure-actuator-autoconfigure'
   type: boolean
   default: true
+- name: release_springcloudazurenativereachability
+  displayName: 'spring-cloud-azure-native-reachability'
+  type: boolean
+  default: true
```
Comment (saragluna) [inline sdk/spring/ci.yml:89]: should this default to false?

File: sdk/spring/spring-cloud-azure-native-reachability/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+## 5.0.0-beta.1 (Unreleased)
+
+### Features Added
+
+### Breaking Changes
+
+### Bugs Fixed
+
+### Other Changes
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-native-reachability/CHANGELOG.md:11]: Let's change the change log.

File: sdk/spring/spring-cloud-azure-native-reachability/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      binaries {
+        main {
+          buildArgs('-Djava.security.properties=' + file("src/main/resources/custom.security").absolutePath)
+        }
+      }
+    }
+    bootRun {
+      systemProperty("java.security.properties", file("src/main/resources/custom.security").absolutePath)
+      systemProperty('spring.aot.enabled', 'true')
+    }
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-native-reachability/README.md:113]: @moarychan could you help verify the correctness of this?

Comment (moarychan) [reply sdk/spring/spring-cloud-azure-native-reachability/README.md:113]: OK, will test it later.

**Pull request merged**